infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+        sequencerServer.setSequencerEpoch(0L);
+        RequestMsg request = getRequestMsg(
+                getBasicHeader(true, true),
+                CorfuProtocolSequencer.getSequencerMetricsRequestMsg()
+        );
+
+        ArgumentCaptor<ResponseMsg> responseCaptor = ArgumentCaptor.forClass(ResponseMsg.class);
+        sequencerServer.handleMessage(request, mockChannelHandlerContext, mockServerRouter);
+
+        verify(mockServerRouter).sendResponse(responseCaptor.capture(), any(ChannelHandlerContext.class));
+        ResponseMsg response = responseCaptor.getValue();
+
+        // Assert that the payload has a getSequencerMetricsResponseMsg and that the base
+        // header fields have remained the same
+        assertTrue(compareBaseHeaderFields(request.getHeader(), response.getHeader()));
+        assertTrue(response.getPayload().hasSequencerMetricsResponse());
+
+        // Assert that the SequencerMetricsResponseMsg contains SequencerStatus.READY
+        SequencerMetricsResponseMsg sequencerMetricsResponseMsg = response.getPayload().getSequencerMetricsResponse();
+        assertTrue(sequencerMetricsResponseMsg.hasSequencerMetrics());
+        assertThat(sequencerMetricsResponseMsg.getSequencerMetrics().getSequencerStatus())
+                .isEqualTo(SequencerStatus.READY);
+    }
+
+    /**
+     * Test that when client sends the SequencerMetricsRequestMsg, the server responds with
+     * SequencerMetricsResponseMsg with SequencerStatus.READY as its content.
+     */
+    @Test
+    public void sequencerTrimRequest() {
+        // Bootstrap the Sequencer Server with a stream
+        UUID streamA = UUID.nameUUIDFromBytes("streamA".getBytes());
+        // Construct new tailMaps and create a to BootstrapSequencerRequestMsg
+        Map<UUID, StreamAddressSpace> tailMap = new HashMap<>();
+        long globalTail = 1;
+        // StreamAddressSpace streamAddressSpace = new StreamAddressSpace(Address.NON_ADDRESS,
+        //         Roaring64NavigableMap.bitmapOf(globalTail));
+        StreamAddressSpace streamAddressSpace = mock(StreamAddressSpace.class);
+        tailMap.put(streamA, streamAddressSpace);
+        long bootstrapMsgEpoch = 1;
+
+        SequencerServerCache cache = mock(SequencerServerCache.class);
+        SequencerServer.FactoryHelper mockFactoryHelper = mock(SequencerServer.FactoryHelper.class);
+        when(mockFactoryHelper.getStreamAddressSpaceMap()).thenReturn(tailMap);
+        // The global log tail is set to 0 when sequencer is bootstrapped with the above tailMap
+        when(mockFactoryHelper.getGlobalLogTail()).thenReturn(globalTail);
+        when(mockFactoryHelper.getSequencerServerCache(anyInt(), anyLong())).thenReturn(cache);
+        sequencerServer = new SequencerServer(mockServerContext, mockFactoryHelper);
+
+        // Send a request with SequencerTrimRequestMsg
+        long trimMark = 0;
+        RequestMsg request = getRequestMsg(
+                getBasicHeader(true, true),
```

Comment:
Should these be equal to false? That's how the message is sent in the client.

Reply:
Done, Thank you.